keyboards/skyloong/gk75/q2/q2.c

```diff
@@ -0,0 +1,294 @@
+                   RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+                 }
+              }
+         } break;
+
+     case 1:{
+       if (L_MAC) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 255, 255, 255);
+            if (!rgb_matrix_get_flags()) {
+               RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+            }
+            }else{
+                if (!rgb_matrix_get_flags()) {
+                   RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+                 }
+              }
+         } break;
+
+      default:{
+         if (!rgb_matrix_get_flags()) {
+            RGB_MATRIX_INDICATOR_SET_COLOR(WIN_MOD_INDEX, 0, 0, 0);
+            RGB_MATRIX_INDICATOR_SET_COLOR(MAC_MOD_INDEX, 0, 0, 0);
+         }
+      }
+    }
+    return false;
+}
+#endif
+
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    if (!process_record_user(keycode, record)) {
+        return false;
+    }
+    switch (keycode) {
+#    ifdef RGB_MATRIX_ENABLE
+       case RGB_TOG:
+           if (record->event.pressed) {
+               switch (rgb_matrix_get_flags()) {
+                    case LED_FLAG_ALL: {
+                        rgb_matrix_set_flags(LED_FLAG_NONE);
+                        rgb_matrix_set_color_all(0, 0, 0);
+                    } break;
+                    default: {
+                        rgb_matrix_set_flags(LED_FLAG_ALL);
+                    } break;
+            }
+        }
+        return false;
+
+       case RGB_VAI:
+       rgb_matrix_set_flags(LED_FLAG_ALL);
+       return true;
+#   endif
+     case MO(1):
+      if (record->event.pressed) {
+       FN_ON = 1;
+      } else {
+       FN_ON = 0;
+      }
+      return true;
+
+     case MO(2):
+      if (record->event.pressed) {
+       FN_WIN = 1;
+      } else {
+       FN_WIN = 0;
+      }
+      return true;
+
+     case MO(3):
+      if (record->event.pressed) {
+       FN_MAC = 1;
+      } else {
+       FN_MAC = 0;
+      }
+      return true;
+
+     case KC_LGUI:
+      if (FN_WIN | FN_MAC | FN_ON){
+          if ( record->event.pressed){
+             WIN_LOCK = !WIN_LOCK ; //change win lock state
+            }
+          if (!WIN_LOCK) {
+             return false; //windows key locked do nothing
+            }
+        }
+      if (WIN_LOCK) {
+             return false; //windows key locked do nothing
+            }
+      return true;  // continue all further processing of this key
+
+     case TO(0):
+      if (record->event.pressed) {
+       L_WIN = 1;
+       set_single_persistent_default_layer(0); // Save default layer 0 to eeprom
+      } else {
+       L_WIN = 0;
+      }
+      return true; // continue all further processing of this key
+
+     case TO(1):
+      if (record->event.pressed) {
+       L_MAC = 1;
+       set_single_persistent_default_layer(1);  //Save default layer 1 to eeprom
+      } else {
+       L_MAC = 0;
+      }
+       return true;
```

Comment:
All of this should be removed, as we do not want this sort of logic at the keyboard level.

* `TO(x)` should never be changed to call `set_single_persistent_default_layer`.
* `layer_state` variable could be used in `rgb_matrix_indicators_advanced_kb`  (or maybe layer_state_set_kb)

Reply:
OK, I modified it. Would you please check it again @zvecr